x-pack/plugins/ml/server/models/job_validation/validate_time_range.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ES_FIELD_TYPES } from '../../../../../../src/plugins/data/server';
+import { parseInterval } from '../../../../../legacy/plugins/ml/common/util/parse_interval';
+import { CombinedJob } from '../../../../../legacy/plugins/ml/public/application/jobs/new_job/common/job_creator/configs';
+import { callWithRequestType } from '../../../../../legacy/plugins/ml/common/types/kibana';
+// @ts-ignore
+import { validateJobObject } from './validate_job_object';
+
+interface ValidateTimeRangeMessage {
+  id: string;
+  timeField?: string;
+  minTimeSpanReadable?: string;
+  bucketSpanCompareFactor?: number;
+}
+
+interface TimeRange {
+  start: number;
+  end: number;
+}
+
+const BUCKET_SPAN_COMPARE_FACTOR = 25;
+const MIN_TIME_SPAN_MS = 7200000;
+const MIN_TIME_SPAN_READABLE = '2 hours';
+
+export async function isValidTimeField(callWithRequest: callWithRequestType, job: CombinedJob) {
+  const index = job.datafeed_config.indices.join(',');
+  const timeField = job.data_description.time_field;
+
+  // check if time_field is of type 'date' or 'date_nanos'
+  const fieldCaps = await callWithRequest('fieldCaps', {
+    index,
+    fields: [timeField],
+  });
+
+  let fieldType = fieldCaps.fields[timeField]?.date?.type;
+  if (fieldType === undefined) {
+    fieldType = fieldCaps.fields[timeField]?.date_nanos?.type;
+  }
+  return fieldType === ES_FIELD_TYPES.DATE || fieldType === ES_FIELD_TYPES.DATE_NANOS;
+}
+
+export async function validateTimeRange(
+  callWithRequest: callWithRequestType,
+  job: CombinedJob,
+  timeRange: TimeRange | undefined
+) {
+  const messages: ValidateTimeRangeMessage[] = [];
+
+  validateJobObject(job);
+
+  // check if time_field is a date type
+  if (!(await isValidTimeField(callWithRequest, job))) {
+    messages.push({
+      id: 'time_field_invalid',
+      timeField: job.data_description.time_field,
+    });
+    // if the time field is invalid, skip all other checks
+    return Promise.resolve(messages);
```

Comment:
it doesn't look like the `Promise.resolve` is needed as this function is `async` and so just returning `messages` will have the same effect.
same on line 72

Reply:
Done in https://github.com/elastic/kibana/pull/59017/commits/b92491bcdc167e2a2b546286e0774e713e60abf3